suite-common/wallet-core/src/blockchain/blockchainReducer.ts

```diff
@@ -0,0 +1,170 @@
+
+export const isHttpProtocol = (url: string) => /^https?:\/\//.test(url);
+
+const initialStatePredefined: Partial<BlockchainState> = {};
+
+// fill initial state, those values will be changed by BLOCKCHAIN.UPDATE_FEE action
+export const blockchainInitialState: BlockchainNetworks = networksCompatibility.reduce(
+    (state, network) => {
+        if (network.accountType) return state;
+        state[network.symbol] = {
+            connected: false,
+            explorer: network.explorer,
+            blockHash: '0',
+            blockHeight: 0,
+            version: '0',
+            backends:
+                network.symbol === 'regtest'
+                    ? {
+                          selected: 'blockbook',
+                          urls: {
+                              blockbook: ['http://localhost:19121'],
+                          },
+                      }
+                    : {},
+        };
+        return state;
+    },
+    initialStatePredefined as BlockchainState,
+);
+
+const connect = (draft: BlockchainState, info: BlockchainInfo) => {
+    const network = getNetwork(info.coin.shortcut.toLowerCase());
+    if (!network) return;
+
+    const isHttp = isHttpProtocol(info.url); // can use dynamic backend url settings
+
+    draft[network.symbol] = {
+        url: info.url,
+        explorer: {
+            tx: `${
+                isHttp
+                    ? info.url + getBlockExplorerUrlSuffix(network.explorer.tx)
+                    : network.explorer.tx
+            }`,
+            account: `${
+                isHttp
+                    ? info.url + getBlockExplorerUrlSuffix(network.explorer.account)
+                    : network.explorer.account
+            }`,
+        },
+        connected: true,
+        blockHash: info.blockHash,
+        blockHeight: info.blockHeight,
+        version: info.version,
+        backends: draft[network.symbol].backends,
+    };
+
+    delete draft[network.symbol].error;
+    delete draft[network.symbol].reconnection;
+};
+
+const error = (draft: BlockchainState, symbol: string, error: string) => {
+    const network = getNetwork(symbol.toLowerCase());
+    if (!network) return;
+
+    draft[network.symbol] = {
+        ...draft[network.symbol],
+        connected: false,
+        explorer: network.explorer,
+        error,
+    };
+    delete draft[network.symbol].url;
+};
+
+const update = (draft: BlockchainState, block: BlockchainBlock) => {
+    const network = getNetwork(block.coin.shortcut.toLowerCase());
+    if (!network) return;
+
+    draft[network.symbol] = {
+        ...draft[network.symbol],
+        blockHash: block.blockHash,
+        blockHeight: block.blockHeight,
+    };
+};
+
+export const prepareBlockchainReducer = createReducerWithExtraDeps(
+    blockchainInitialState,
+    (builder, extra) => {
+        builder
+            .addCase(extra.actionTypes.storageLoad, extra.reducers.storageLoadBlockchain)
```

Comment:
nit based on the latest comment from @Nodonisko in my fiat-rates PR [1]

_"I would probably prefer to have all cases that use extra at the end, like we did it in other PRs"_

[1] https://github.com/trezor/trezor-suite/pull/6118#discussion_r956146664

Reply:
https://github.com/trezor/trezor-suite/pull/6140/commits/5a38c9aea55e9811f7fecdf7679ca1a86081f8ec